src/lib/file_operations.js

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+/* eslint-disable no-console */
+
+const fs = require("fs")
+const { generateMarkdownFromJson } = require("./markdown_generators")
+const cliProgress = require("cli-progress")
+const progressBar = new cliProgress.SingleBar(
+  {},
+  cliProgress.Presets.shades_classic
+)
+let totalDirectories = 0
+let directoriesScanned = 0
+let filesProcessed = 0
+
+const scanCourses = (source, destination) => {
+  /*
+    This function scans the source directory for course folders
+  */
+  fs.readdir(source, (err, directories) => {
```

Comment:
Can we use `readdirSync` here and below? Alternatively we can change these to async functions and use util.promisify, but I'm not sure that would be necessary here.

Reply:
Or [fsPromises](https://nodejs.org/api/fs.html#fs_fspromises_readdir_path_options) in place of `util.promisify`

Reply:
Punting on this, per our discussion about the memory leak issue with turndown.